### DISCUSSION TITLE
Data migration to v2.0.0 failed: (FunctionClauseError) no function clause matching in String.trim/1

### DISCUSSION THREAD

**Discussion by dmitrijszaharovs** [plausible/analytics#3146] [Self-Hosted Support] [answered]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

Following the instruction on migrating to v.2.0.0 (https://github.com/plausible/analytics/releases/tag/v2.0.0), 
I've tried to migrate my data to v.2.0.0 but failed.
To have a clean environment, I have created a new Ubuntu VM, started my old docker-compose configuration with adjusted version to `plausible/analytics:v2.0.0`, got the `plausible` container ID and run a command:
```
docker exec 44f16a08816c bin/plausible rpc Plausible.DataMigration.NumericIDs.run
```
...as mentioned in the manual.
As a result - got an error: 
```
root@aegusanix2vm:/opt/hosting# docker exec 44f16a08816c bin/plausible rpc Plausible.DataMigration.NumericIDs.run
    check-replicas Done!

------------------------------------------------------------------------------
    list-partitions Done!

------------------------------------------------------------------------------
Got the following migration settings:

  - max_threads: 16
  - table_settings: SETTINGS index_granularity = 8192
  - db url: http://plausible_events_db:8123/plausible_events_db
  - cluster?: false
  - partitions to do: ["202212", "202301", "202302", "202303", "202304", "202305", "202306", "202307"]
  - start from: 202212
  - stop at:

Running drop-events-v2
  -> Query: DROP TABLE IF EXISTS events_v2  SYNC

Execute? [y/N]: ** (FunctionClauseError) no function clause matching in String.trim/1

    The following arguments were given to String.trim/1:

        # 1
        :eof

    (elixir 1.14.3) lib/string.ex:1271: String.trim/1
    lib/plausible/data_migration/numeric_ids.ex:6: Plausible.DataMigration.NumericIDs.confirm/3
    lib/plausible/data_migration/numeric_ids.ex:105: Plausible.DataMigration.NumericIDs.run/1
    nofile:1: (file)
    (stdlib 4.2) erl_eval.erl:748: :erl_eval.do_apply/7
```

### Expected behavior

Data migration finished without errors and success message is displayed.

### Screenshots

![image](https://github.com/plausible/analytics/assets/62420667/a3bcb7dc-57d2-4547-94f3-cfe829724898)

### Environment

```markdown
- OS: Debian bullseye/sid
- Browser: n/a
- Browser Version: n/a
```

**Comment by ruslandoga** [comment, marked as answer]
👋 @dmitrijszaharovs 

You'd need to add `-ti` flags to `docker exec` as its default behaviour is different from `docker compose exec` used in the instructions.

**Reply by ruslandoga** [reply]
Please see https://github.com/plausible/analytics/discussions/3035 for more info.

**Reply by dmitrijszaharovs** [reply]
Great! Thanks! It works now and data migrated successfully.
